sapl-lang/src/main/java/io/sapl/interpreter/InputStreamHelper.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017-2024 Dominic Heutelbeck (dominic@heutelbeck.com)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.sapl.interpreter;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.ByteOrderMark;
+import org.apache.commons.io.input.BOMInputStream;
+import org.apache.commons.io.input.ReaderInputStream;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class InputStreamHelper {
+
+    public static InputStream detectAndConvertEncodingOfStream(InputStream policyInputStream) throws IOException {
+
+        BOMInputStream bomIn = BOMInputStream
+                .builder().setInputStream(policyInputStream).setByteOrderMarks(ByteOrderMark.UTF_16LE,
+                        ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_32LE, ByteOrderMark.UTF_32BE, ByteOrderMark.UTF_8)
+                .get();
+
+        if (!bomIn.hasBOM()) {
+            // InputStream without BOM is treated as UTF-8
+            return getUtf8InputStream(new InputStreamReader(bomIn, StandardCharsets.UTF_8));
```

Comment:
See SonarCloud. Use Static default encoding Stings everywhere.

Reply:
UTF-32 and similar are not part of the standard charset

https://bugs.openjdk.org/browse/JDK-8310251

Rest has been fixed.